packages/stylelint-plugin/src/util/kaizenTokens.spec.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  getReplacementForDeprecatedOrRemovedToken,
+  isKaizenTokenDeprecatedOrRemoved,
+  kaizenTokensByName,
+} from "./kaizenTokens"
+
+describe("kaizenTokens", () => {
+  describe(getReplacementForDeprecatedOrRemovedToken.name, () => {
+    test("never returns another deprecated token across all Kaizen tokens", () => {
+      Object.entries(kaizenTokensByName).forEach(([key, value]) => {
+        if (!value) return
+        const replacement = getReplacementForDeprecatedOrRemovedToken(
+          value.name
+        )
+        if (!replacement) return
+        if (isKaizenTokenDeprecatedOrRemoved(replacement.name)) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `Kaizen token: ${value.name}, is deprecated and has a replacement of ${replacement.name}, but the replacement is also deprecated`
```

Comment:
This warning is pretty good at describing what the expectation is.

Reply:
I gave this a shot but it ended up with ~660 more tests - not necessarily a bad thing.

I interpreted this test as more of a _property based_ test, meaning, an assertion on the function over the input domain (not every possible string obviously, but just the ones that are the most likely which are all Kaizen token names).
I'm not sure what the most reasonable thing to do for Jest is. Property based happens in other languages/ecosystems i think it's a single test.
That being said, you could interpret this in the same way as the list of inputs + outputs in the spec that tests the whole plugin.
Because we're not hand coding the list of test cases, i.e. hand picking and declaring the inputs + outputs, i think it makes sense to keep this as a single test.